### PR TITLE
Adding the ChartLabelSettings and auxiliary classes to allow specifyi…

### PR DESCRIPTION
**Pull request by jlalvarezwf**
Adding the ChartLabelSettings and auxiliary classes to allow specifying labels for a chart.

### CODE REVIEW

File: src/app/showcase/components/chart/showcase-chart.component.ts [hunk 1]
```diff
@@ -87,7 +89,38 @@ export class ShowcaseChartComponent {
...
+			const currentPercentage = context.dataset.data[context.dataIndex] * 100 / arraySum(dataArr);
+
+			return currentPercentage >= 5;
+		}
+		this.pieChartLabelSettings.position.display = displayFunction;
+		this.pieChartLabelSettings.labelColors = new ChartLabelColor(undefined, 'black', undefined, 5, 1, 0.8);
+		const fontFamily = 'Courier, Arial Unicode MS, Arial, sans-serif';
+		this.pieChartLabelSettings.chartLabelFont = new ChartLabelFont(undefined, fontFamily, 16, undefined, 'bold', 0.8);
+		this.pieChartLabelSettings.chartLabelPadding = new ChartLabelPadding(undefined, 1, 1, 1, 1);
+		// this.pieChartLabelSettings.chartLabelText = new ChartLabelText(undefined, 'rgb(200, 200, 200)', 1, 0.2, 'black');
```
Comment (cokecc) [inline src/app/showcase/components/chart/showcase-chart.component.ts]: delete this commented code

File: src/app/showcase/components/chart/showcase-chart.component.ts [hunk 2]
```diff
@@ -87,7 +89,38 @@ export class ShowcaseChartComponent {
 
 		this.dataDoughnut.push(new ChartItem('', [36, 23, 42, 52], '', '', true, true, false, 3));
 
-		const pieChartItem = new ChartItem('', [36, 23, 42, 52], '', '#778899', true, true, false, 3);
+		this.pieChartLabelSettings = new ChartLabelSettings();
+		this.pieChartLabelSettings.position = new ChartLabelPosition();
+		this.pieChartLabelSettings.position.clip = false; // to avoid showing part of the label, set clip = true
+
+		const arraySum = (dataArr: Array<number>): number => {
```
Comment (vinyulis) [inline src/app/showcase/components/chart/showcase-chart.component.ts]: It is better to use reduce() to sum the content of an array

Comment (jlalvarezwf) [reply src/app/showcase/components/chart/showcase-chart.component.ts]: Done!

File: src/app/showcase/components/chart/showcase-chart.component.ts (hunk 1 above)
Comment (jlalvarezwf) [reply src/app/showcase/components/chart/showcase-chart.component.ts]: Done!

**Pull request merged**